lib/templateFactory.js

```diff
@@ -80,32 +81,60 @@ class TemplateFactory extends BaseFactory {
     }
 
     /**
-     * Get a the latest template by config
+     * Get a the latest template by config using the full template name
      * @method getTemplate
-     * @param  {Object}     config               Config object
-     * @param  {String}     config.name          The template name
-     * @param  {String}     config.version       Version of the template
-     * @param  {String}     [config.label]       Label of template
-     * @return {Promise}                         Resolves template model or undefined if not found
+     * @param  {String}     fullTemplateName    Name of the template and the version or tag (e.g. chef/publish@1.2.3)
+     * @return {Promise}                        Resolves template model or null if not found
      */
-    getTemplate(config) {
-        // get all templates with the same name
-        return super.list({ params: { name: config.name } })
+    getTemplate(fullTemplateName) {
+        const match = TEMPLATE_NAME_REGEX.exec(fullTemplateName);
+        const templateName = match[TEMPLATE_NAME_MATCH];
+        const versionOrTag = match[VERSION_OR_TAG_MATCH];
+        const isVersion = VERSION_REGEX.exec(versionOrTag);
+
+        // If tag is passed in, get the version from the tag
+        if (versionOrTag && !isVersion) {
+            // Lazy load factory dependency to prevent circular dependency issues
+            // eslint-disable-next-line global-require
+            const TemplateTagFactory = require('./templateTagFactory');
+            const templateTagFactory = TemplateTagFactory.getInstance();
+
+            // Get a template tag
+            return templateTagFactory.get({
+                name: templateName,
+                tag: versionOrTag
+            })
+            .then((templateTag) => {
+                // Return null if no template tag exists
+                if (!templateTag) {
+                    return null;
+                }
+
+                // Get a template using the exact template version
+                return super.get({
+                    name: templateName,
+                    version: templateTag.version
+                });
+            });
+        }
+
+        // Get all templates with the same name
+        return super.list({ params: { name: templateName } })
             .then((templates) => {
-                // get templates that have version prefix as config.version
+                // If no version provided, return the most recently published template
+                if (!versionOrTag) {
+                    return templates[0];
+                }
+
+                // Get templates that have versions beginning with the version given
                 const filtered = templates.filter(template =>
-                    template.version.startsWith(config.version));
+                    template.version.startsWith(versionOrTag));
```

Comment:
Should have a check before the list. If it's a exact version then just use a `get`